include/flashinfer/gemm/group_gemm_sm90.cuh

```diff
@@ -121,8 +121,7 @@ cudaError_t CutlassSegmentGEMMSM90Run(void* float_buffer, size_t float_buffer_si
 
       cutlass::KernelHardwareInfo hw_info;
       cudaGetDevice(&hw_info.device_id);
-      hw_info.sm_count =
-          cutlass::KernelHardwareInfo::query_device_multiprocessor_count(hw_info.device_id);
+      hw_info.sm_count = num_ctas;
```

Comment:
I tried nsys profiler and it turns out this value can't control the number of SMs this kernel used. 
A more fundamental approach might be using [green context](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__GREEN__CONTEXTS.html#group__CUDA__GREEN__CONTEXTS).